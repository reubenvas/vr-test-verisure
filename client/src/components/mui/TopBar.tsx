import React from 'react';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { Grid } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) => createStyles({
    root: {
        flexGrow: 1,
    },
    menuButton: {
        marginRight: theme.spacing(2),
    },
    popoutButton: {
        position: 'absolute',
        width: '100vw',
        justifyContent: 'center',
        left: 0,
        right: 0,
        top: 25,
    },
}));

const DenseAppBar = (): React.ReactElement<{}> => {
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <AppBar position="fixed">
                <Toolbar variant="dense" >
                    <Grid container>
                        <Grid item>
                            <Typography variant="h4" color="inherit" component='h1'>
                                Visual regression testing
                            </Typography>
                        </Grid>
                        <Grid item justify='center' container alignItems='center' className={classes.popoutButton}>
                            <Button variant="contained" className={classes.menuButton} color="secondary">Add scenario</Button>
                        </Grid>
                    </Grid>
                </Toolbar>
            </AppBar>
        </div>
    );
};

export default DenseAppBar;
