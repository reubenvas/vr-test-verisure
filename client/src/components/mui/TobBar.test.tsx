import React from 'react';
import renderer from 'react-test-renderer';
import TopBar from './TopBar';

describe('DarkButton component', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<TopBar />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
