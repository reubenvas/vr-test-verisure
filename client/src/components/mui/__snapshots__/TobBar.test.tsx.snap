// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkButton component snapshot renders 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <h1
            className="MuiTypography-root MuiTypography-h4 MuiTypography-colorInherit"
          >
            Visual regression testing
          </h1>
        </div>
        <div
          className="MuiGrid-root makeStyles-popoutButton-3 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Add scenario
            </span>
          </button>
          <div />
        </div>
      </div>
    </div>
  </header>
</div>
`;
